### PR TITLE
[CNDIT-1340] Investigation Notification/Case Answer Logic Update

### PR DESCRIPTION
**Pull request by rshanmugam-enquizit**
### Notes
This PR includes the logic update for the `investigation_notifications` and `investigation_case_answer` json from the `sp_investigation_event` stored proc results. This also includes the changes needed for the `investigation_case_answer` to directly save the data to the rdb instead of the sync connector.

Below is the new env variables for the investigation service to run in local (update the db host, username and password) - 

```
DA_LOG_PATH=logs;DB_ODSE=NBS_ODSE;DB_PASSWORD=$$$;DB_RDB=rdb;DB_URL=$$$;DB_USERNAME=$$$;KAFKA_BOOTSTRAP_SERVER=localhost:9092
```

### JIRA
- https://cdc-nbs.atlassian.net/browse/CNDIT-1340

### Testing
- Unit tests with 90% coverage added (except model classes).

### CODE REVIEW

File: data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java [hunk 1]
```diff
@@ -44,10 +51,39 @@ public InvestigationTransformed transformInvestigationData(Investigation investi
         transformNotificationIds(investigation.getObservationNotificationIds(), objectMapper);
```
Comment (sveselev) [inline data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java:52]: Do we still need processing notification ids and `nrt_investigation_notification`?

**Review by sveselev — APPROVED** [review]

File: data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java [hunk 2]
```diff
@@ -44,10 +52,39 @@ public InvestigationTransformed transformInvestigationData(Investigation investi
...
         transformObservationIds(investigation.getObservationNotificationIds(), investigationTransformed, objectMapper);
         transformInvestigationConfirmationMethod(investigation.getInvestigationConfirmationMethod(), investigationTransformed, objectMapper);
+        processInvestigationPageCaseAnswer(investigation.getInvestigationCaseAnswer(), objectMapper);
```
Comment (alog-enquizit) [inline data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java:55]: Page case answers bypass the connectors while notfications will still use the Kafka topics/Sink connector path right.

File: data-reporting-service/investigation-service/src/test/java/gov/cdc/etldatapipeline/investigation/InvestigationDataProcessingTests.java [hunk 3]
```diff
@@ -38,6 +48,7 @@ public class InvestigationDataProcessingTests {
     private static final String CONFIRMATION_TOPIC = "confirmationTopic";
     private static final String OBSERVATION_TOPIC = "observationTopic";
     private static final String NOTIFICATION_TOPIC = "notificationTopic";
+    private static final String NOTIFICATIONS_TOPIC = "notificationsTopic";
```
Comment (upasanapattnaik-eq) [inline data-reporting-service/investigation-service/src/test/java/gov/cdc/etldatapipeline/investigation/InvestigationDataProcessingTests.java:51]: NOTIFICATION_TOPIC on line 50 will be removed as part of the observation_notification_ids, correct?

**Review by upasanapattnaik-eq — APPROVED** [review]
Tested case answer functionality in Alabama env. LGTM :+1: 
Thank you for implementing!

File: data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java (hunk 1 above)
Comment (rshanmugam-enquizit) [reply data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java:52]: for now, yes - https://cdc-nbs.atlassian.net/wiki/spaces/NM/pages/533790760/005+investigation#Table---investigation_notification-pHASE-OUT but we need to remove it once a decision is made

File: data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java (hunk 2 above)
Comment (rshanmugam-enquizit) [reply data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java:55]: right!

**Pull request merged**